core/embed/bootloader/emulator.c

```diff
@@ -44,26 +74,61 @@ __attribute__((noreturn)) int main(int argc, char **argv) {
     (void)ret;
   }
 
-  if (argc == 2) {
-    if (argv[1][0] == 's') {
-      // Run the firmware
-      stay_in_bootloader_flag = STAY_IN_BOOTLOADER_FLAG;
-    }
+  int opt;
+  bool display_error = false;
+  uint8_t set_variant = 0xff;
+  uint8_t color_variant = 0;
+  uint8_t bitcoin_only = 0;
+  while ((opt = getopt(argc, argv, "slec:b:")) != -1) {
+    switch (opt) {
+      case 's':
+        stay_in_bootloader_flag = STAY_IN_BOOTLOADER_FLAG;
+        break;
+      case 'e':
+        display_error = true;
+        break;
+      case 'c':
+        set_variant = 1;
+        color_variant = atoi(optarg);
+        break;
+      case 'b':
+        set_variant = 1;
+        bitcoin_only = atoi(optarg);
+        break;
 #ifdef USE_OPTIGA
-    else if (argv[1][0] == 'l') {
-      // write bootloader-lock secret
-      secret_write_header();
-    }
+      case 'l':
+        // write bootloader-lock secret
+        secret_write_header();
+        break;
 #endif
-  } else if (argc == 4) {
-    display_init();
-    display_backlight(180);
-    screen_fatal_error_rust(argv[1], argv[2], argv[3]);
-    display_refresh();
-    ui_click();
-    exit(0);
+      default:
+        printf("Unrecognized argument or something\n");
+        usage();
```

Comment:
37412464f81b7b138c399e7bb98036130a2d1172 removes the print completely, because a print like that is emitted by getopt itself, as I noticed :)
656b1bf41a1b01b016eee0e11a54380c1f3a858d removes the special handling for `-h` because the default case is good enough when `h` is recognized. there's this exit(1) which is a tiny bit ugly but honestly who cares